src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -684,20 +679,49 @@ private void PopulateSessionProperties(OpenIdConnectMessage message, Authenticat
         }
 
         /// <summary>
-        /// Redeems the authorization code for tokens at the token endpoint
+        /// Redeems the authorization code for tokens at the token endpoint.
+        /// 
+        /// The authorization response may contain error while its status code doesn't indicate failure. In this case
```

Comment:
we decided from the spec that this isn't true, the status code should be 400 for errors